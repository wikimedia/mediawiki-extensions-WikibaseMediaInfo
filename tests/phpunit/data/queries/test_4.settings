{
	"title": "Entities found, no lang fallbacks, filter applied",
	"term": "some search term filetype:bitmap",
	"entityIdsForTerm": {
		"some search term": [
			"Q1",
			"Q2"
		]
	},
	"defaultProperties": [
		"P1",
		"P2"
	],
	"userLanguage": "en",
	"settings": {
		"statement": 20.0,
		"statement-discount": 0.1,
		"caption": 10.0,
		"caption-fallback-discount": 0.5,
		"all": 0.9,
		"all.plain": 0.8,
		"category": 0.7,
		"title": 0.6,
		"redirect.title": 0.5,
		"text": 0.4,
		"auxiliary_text": 0.3,
		"file_text": 0.2
	}
}
