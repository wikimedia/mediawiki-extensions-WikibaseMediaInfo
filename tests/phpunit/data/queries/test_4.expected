{
    "bool": {
        "filter": [
            {
                "bool": {
                    "should": [
                        {
                            "multi_match": {
                                "fields": [
                                    "all",
                                    "all.plain"
                                ],
                                "query": "some search term",
                                "operator": "and"
                            }
                        },
                        {
                            "bool": {
                                "should": [
                                    {
                                        "match": {
                                            "statement_keywords": {
                                                "query": "P1=Q1"
                                            }
                                        }
                                    },
                                    {
                                        "match": {
                                            "statement_keywords": {
                                                "query": "P2=Q1"
                                            }
                                        }
                                    },
                                    {
                                        "match": {
                                            "statement_keywords": {
                                                "query": "P1=Q2"
                                            }
                                        }
                                    },
                                    {
                                        "match": {
                                            "statement_keywords": {
                                                "query": "P2=Q2"
                                            }
                                        }
                                    },
                                    {
                                        "match": {
                                            "statement_keywords": {
                                                "query": "P1=Q999999"
                                            }
                                        }
                                    },
                                    {
                                        "match": {
                                            "statement_keywords": {
                                                "query": "P2=Q999999"
                                            }
                                        }
                                    }
                                ]
                            }
                        }
                    ]
                }
            },
            {
                "match": {
                    "file_media_type": {
                        "query": "bitmap"
                    }
                }
            }
        ],
        "should": [
            [
                {
                    "function_score": {
                        "query": {
                            "bool": {
                                "should": [
                                    {
                                        "function_score": {
                                            "query": {
                                                "bool": {
                                                    "should": [
                                                        {
                                                            "bool": {
                                                                "should": [
                                                                    {
                                                                        "multi_match": {
                                                                            "query": "some search term",
                                                                            "boost": 1,
                                                                            "minimum_should_match": 1,
                                                                            "type": "most_fields",
                                                                            "fields": [
                                                                                "descriptions.en.plain^1"
                                                                            ]
                                                                        }
                                                                    },
                                                                    {
                                                                        "multi_match": {
                                                                            "query": "some search term",
                                                                            "boost": 1,
                                                                            "minimum_should_match": 1,
                                                                            "type": "most_fields",
                                                                            "fields": [
                                                                                "title^3",
                                                                                "title.plain^1"
                                                                            ]
                                                                        }
                                                                    },
                                                                    {
                                                                        "multi_match": {
                                                                            "query": "some search term",
                                                                            "boost": 1,
                                                                            "minimum_should_match": 1,
                                                                            "type": "most_fields",
                                                                            "fields": [
                                                                                "category^3",
                                                                                "category.plain^1"
                                                                            ]
                                                                        }
                                                                    },
                                                                    {
                                                                        "multi_match": {
                                                                            "query": "some search term",
                                                                            "boost": 1,
                                                                            "minimum_should_match": 1,
                                                                            "type": "most_fields",
                                                                            "fields": [
                                                                                "redirect.title^3",
                                                                                "redirect.title.plain^1"
                                                                            ]
                                                                        }
                                                                    },
                                                                    {
                                                                        "multi_match": {
                                                                            "query": "some search term",
                                                                            "boost": 1,
                                                                            "minimum_should_match": 1,
                                                                            "type": "most_fields",
                                                                            "fields": [
                                                                                "suggest"
                                                                            ]
                                                                        }
                                                                    }
                                                                ]
                                                            }
                                                        },
                                                        {
                                                            "bool": {
                                                                "should": [
                                                                    {
                                                                        "multi_match": {
                                                                            "query": "some search term",
                                                                            "boost": 1,
                                                                            "minimum_should_match": 1,
                                                                            "type": "most_fields",
                                                                            "fields": [
                                                                                "heading^3",
                                                                                "heading.plain^1"
                                                                            ]
                                                                        }
                                                                    },
                                                                    {
                                                                        "multi_match": {
                                                                            "query": "some search term",
                                                                            "boost": 1,
                                                                            "minimum_should_match": 1,
                                                                            "type": "most_fields",
                                                                            "fields": [
                                                                                "auxiliary_text^3",
                                                                                "auxiliary_text.plain^1"
                                                                            ]
                                                                        }
                                                                    },
                                                                    {
                                                                        "multi_match": {
                                                                            "query": "some search term",
                                                                            "boost": 1,
                                                                            "minimum_should_match": 1,
                                                                            "type": "most_fields",
                                                                            "fields": [
                                                                                "text^3",
                                                                                "text.plain^1"
                                                                            ]
                                                                        }
                                                                    },
                                                                    {
                                                                        "multi_match": {
                                                                            "query": "some search term",
                                                                            "boost": 1,
                                                                            "minimum_should_match": 1,
                                                                            "type": "most_fields",
                                                                            "fields": [
                                                                                "file_text^3",
                                                                                "file_text.plain^1"
                                                                            ]
                                                                        }
                                                                    }
                                                                ]
                                                            }
                                                        }
                                                    ]
                                                }
                                            },
                                            "functions": [
                                                {
                                                    "script_score": {
                                                        "script": {
                                                            "source": "max(0, ln(_score))",
                                                            "lang": "expression"
                                                        }
                                                    }
                                                }
                                            ]
                                        }
                                    },
                                    {
                                        "function_score": {
                                            "query": {
                                                "match_explorer": {
                                                    "type": "unique_terms_count",
                                                    "query": {
                                                        "match": {
                                                            "text": {
                                                                "query": "some search term"
                                                            }
                                                        }
                                                    }
                                                }
                                            },
                                            "functions": [
                                                {
                                                    "script_score": {
                                                        "script": {
                                                            "source": "ln(1 \/ max(1.25, _score))",
                                                            "lang": "expression"
                                                        }
                                                    }
                                                }
                                            ]
                                        }
                                    }
                                ]
                            }
                        },
                        "functions": [
                            {
                                "script_score": {
                                    "script": {
                                        "source": "exp(_score)",
                                        "lang": "expression"
                                    }
                                }
                            }
                        ]
                    }
                },
                {
                    "dis_max": {
                        "queries": [
                            {
                                "match": {
                                    "statement_keywords": {
                                        "query": "P1=Q1",
                                        "boost": 1
                                    }
                                }
                            },
                            {
                                "match": {
                                    "statement_keywords": {
                                        "query": "P2=Q1",
                                        "boost": 1
                                    }
                                }
                            },
                            {
                                "match": {
                                    "statement_keywords": {
                                        "query": "P1=Q2",
                                        "boost": 0.75
                                    }
                                }
                            },
                            {
                                "match": {
                                    "statement_keywords": {
                                        "query": "P2=Q2",
                                        "boost": 0.75
                                    }
                                }
                            },
                            {
                                "match": {
                                    "statement_keywords": {
                                        "query": "P1=Q999999",
                                        "boost": 0.56666666666666665
                                    }
                                }
                            },
                            {
                                "match": {
                                    "statement_keywords": {
                                        "query": "P2=Q999999",
                                        "boost": 0.56666666666666665
                                    }
                                }
                            }
                        ]
                    }
                },
                {
                    "bool": {
                        "must": [
                            {
                                "multi_match": {
                                    "fields": [
                                        "all_near_match^2",
                                        "all_near_match.asciifolding^1.5"
                                    ],
                                    "query": "some search term",
                                    "operator": "and"
                                }
                            }
                        ],
                        "filter": [
                            {
                                "terms": {
                                    "namespace": [
                                        0,
                                        14
                                    ]
                                }
                            }
                        ],
                        "boost": 1
                    }
                }
            ]
        ]
    }
}